typeset/converter.js

```diff
@@ -239,7 +243,61 @@ async function injectCoverageCollection (page) {
   }
 }
 
+const highlightCodeElements = (log, inputPath) => {
+  let inputFile
+  const highlightPath = inputPath.replace('baked', 'highlight')
+
+  if (!fileExists.sync(inputPath)) {
+    log.error(`Input XHTML file not found: "${inputPath}"`)
+    return STATUS_CODE.ERROR
+  }
+
+  try {
+    const data = fs.readFileSync(inputPath, 'utf8')
+    inputFile = data.toString()
+  } catch (err) {
+    log.error(`Error: "${err}`)
+    return STATUS_CODE.ERROR
+  }
+
+  const dom = new JSDOM(inputFile, {
+    contentType: 'application/xhtml+xml'
+  })
+
+  const preTagElements = [...dom.window.document.querySelectorAll('pre')]
+  preTagElements.forEach(pre => {
+    const langClass = pre.getAttribute('lang')
+    const highlightedCode = hljs.highlight(langClass, pre.textContent).value
+    pre.innerHTML = '<tempElement xmlns="http://www.w3.org/1999/xhtml">' + highlightedCode + '</tempElement>'
+    if (pre.childNodes.length !== 1) {
+      throw new Error('Bug: Should have had one child')
+    }
+    const tempElement = pre.firstElementChild
+    tempElement.remove()
+    const children = [...tempElement.childNodes]
+    children.forEach(c => {
+      pre.append(c)
+    })
+  })
+
+  const highlightedOutput = serialize(dom.window.document.documentElement)
+  fs.writeFile(highlightPath, highlightedOutput, (err) => {
+    if (err) {
+      log.error(`Error:"${err}"`)
+      return STATUS_CODE.ERROR
+    }
+    log.info('Code snippets have been highlighted')
+  })
+
+  return highlightPath
+}
+
+// TODO:
+// address all codecov comments on github
+// figure out where to add the style sheet
```

Comment:
@philschatz The stylesheet is hosted on `cdnjs`, `jsdelivr`, and `unpkg`. 
I was thinking that I could just inject the link in the head as one of the steps in the `highlightCodeElements` function. If that's the right move, which host do you think I should use? 

https://highlightjs.org/download/

Reply:
Response: We'll make our own theme